packages/transaction-controller/src/utils/utils.ts

```diff
@@ -183,3 +184,31 @@ export function padHexToEvenLength(hex: string) {
 
   return prefix + evenData;
 }
+
+/**
+ * Calculate the percentage difference between two hex values and determine if it is within a threshold.
+ *
+ * @param value1 - The first hex value.
+ * @param value2 - The second hex value.
+ * @param threshold - The percentage threshold for considering the values equal.
+ * @returns Whether the percentage difference is within the threshold.
+ */
+export function isPercentageDifferenceWithinThreshold(
+  value1: Hex,
+  value2: Hex,
+  threshold: number,
+): boolean {
+  const bnValue1 = new BigNumber(value1);
+  const bnValue2 = new BigNumber(value2);
+
+  if (bnValue1.isZero() && bnValue2.isZero()) {
+    return true;
+  }
+
+  const difference = bnValue1.minus(bnValue2).abs();
+  const average = bnValue1.plus(bnValue2).dividedBy(2);
+
+  const percentageDifference = difference.dividedBy(average).multipliedBy(100);
```

Comment:
Can we use BigInt here instead? We support it already on extension and mobile. That would let us avoid adding a dependency, which would complicate the process of backporting this.

BigInt would truncate the result of the division, rounding down to the nearest integer. But if we want more precision, we can multiple it by 10 ^ [number of digits of precision] first.

e.g if we assume that `difference` and `average` are BigInts:

```suggestion
  const percentageDifference = ((difference * 1000) / average) * 100
```

Reply:
Just updated that suggestion, it had a mistake (forgot to divide by the precision factor at the end). It should be good now.

If the threshold is a float as we, could convert to a float as the final step before division as well.

Reply:
Could we also remove the division entirely by just calculating the upper and lower bounds with the first value via multiplication, and checking the second value falls within that range?

Reply:
Yes, that sounds simpler